### PR TITLE
Removing Reference assets from packages which contain runtime-specific dependencies when targeting Desktop.

### PR DESCRIPTION
**Pull request by joperezr**
Porting https://github.com/dotnet/corefx/pull/33254 to master branch. I cherry-picked the change first and then fixed all of the conflicts regarding the move to using Directory.Build.props over dir.props and such, plus removing all of the package version changes.

cc: @ericstj @natemcmaster

### CODE REVIEW

File: src/Directory.Build.targets [hunk 1]
```diff
@@ -57,4 +57,12 @@
     <Message Text="DumpTargets> $(OutputPath), C=[$(Configuration)], CG=[$(ConfigurationGroup)], OG=[$(OSGroup)], TG=[$(TargetGroup)]" Importance="Low" />
   </Target>
 
+  <!-- This target will be moved into buildtools. Checked in here for now to add validationand in order not to block buildtools ingestion. -->
```
Comment (ericstj) [inline src/Directory.Build.targets]: actually, now arcade

File: src/System.Data.SqlClient/pkg/System.Data.SqlClient.pkgproj [hunk 2]
```diff
@@ -6,14 +6,8 @@
       <SupportedFramework>net461;netcoreapp2.0;uap10.0.16299;$(UAPvNextTFM);$(AllXamarinFrameworks)</SupportedFramework>
     </ProjectReference>
     <ProjectReference Include="..\src\System.Data.SqlClient.csproj" />
-    <HarvestIncludePaths Include="ref/net451;lib/net451;runtimes/win/lib/net451" />
-    <HarvestIncludePaths Include="ref/net46;lib/net46;runtimes/win/lib/net46" />
-    <HarvestIncludePaths Include="ref/netstandard1.2">
```
Comment (ericstj) [inline src/System.Data.SqlClient/pkg/System.Data.SqlClient.pkgproj]: Separate issue.  I bet we could delete some live configurations from this package and harvest them giving the API isn't going to change and we have a shipping package with a version of the not-supported assemblies to harvest.

**Comment by benaadams** [comment]
Warnings
```
...
Microsoft.DotNet.GenFacades.targets(98,5): warning : PDB0018: token 0x06000057: Invalid local constant signature: 'EPSILON' [D:\j\workspace\windows-TGrou---0d2c9ac4\src\System.Numerics.Vectors\src\System.Numerics.Vectors.csproj]
Microsoft.DotNet.GenFacades.targets(98,5): warning : PDB0018: token 0x06000070: Invalid local constant signature: 'FLT_EPSILON' [D:\j\workspace\windows-TGrou---0d2c9ac4\src\System.Numerics.Vectors\src\System.Numerics.Vectors.csproj]
Microsoft.DotNet.GenFacades.targets(98,5): warning : PDB0018: token 0x06000089: Invalid local constant signature: 'epsilon' [D:\j\workspace\windows-TGrou---0d2c9ac4\src\System.Numerics.Vectors\src\System.Numerics.Vectors.csproj]
.dotnet\sdk\2.1.401\Microsoft.Common.CurrentVersion.targets(2110,5):
 warning MSB3245: Could not resolve this reference. Could not locate the assembly "System.Threading.ThreadPool".
 Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
 [D:\j\workspace\windows-TGrou---0d2c9ac4\src\System.Threading.Channels\src\System.Threading.Channels.csproj]
```
Errors
```
error : Cannot harvest package 'System.IO.Pipelines' version '4.5.0' because D:\j\workspace\windows-TGrou---0d2c9ac4\.packages\System.IO.Pipelines\4.5.0 does not exist.  Harvesting is needed to redistribute assets and ensure compatiblity with the previous release.  You can disable this by setting HarvestStablePackage=false. [D:\j\workspace\windows-TGrou---0d2c9ac4\src\System.IO.Pipelines\pkg\System.IO.Pipelines.pkgproj]
error : Cannot harvest package 'System.Net.Http.WinHttpHandler' version '4.5.0' because D:\j\workspace\windows-TGrou---0d2c9ac4\.packages\System.Net.Http.WinHttpHandler\4.5.0 does not exist.  Harvesting is needed to redistribute assets and ensure compatiblity with the previous release.  You can disable this by setting HarvestStablePackage=false. [D:\j\workspace\windows-TGrou---0d2c9ac4\src\System.Net.Http.WinHttpHandler\pkg\System.Net.Http.WinHttpHandler.pkgproj]
error : Cannot have Exclude=Compile dependencies when targeting a desktop TFM.
 System.Memory. You can exclude the reference asset in the package by setting the ExcludeReferenceAssets property to true in your project. [D:\j\workspace\windows-TGrou---0d2c9ac4\src\System.Numerics.Tensors\pkg\System.Numerics.Tensors.pkgproj]
error : Cannot have Exclude=Compile dependencies when targeting a desktop TFM.
 System.Runtime.WindowsRuntime. You can exclude the reference asset in the package by setting the ExcludeReferenceAssets property to true in your project. [D:\j\workspace\windows-TGrou---0d2c9ac4\src\System.Runtime.WindowsRuntime.UI.Xaml\pkg\System.Runtime.WindowsRuntime.UI.Xaml.pkgproj]

error : System.Data.SqlClient should support API version 4.4.0.0 on .NETCoreApp,Version=v2.0/win7-x86 but
 System.Data.SqlClient\netstandard\System.Data.SqlClient.dll was found to support 4.6.0.0. [D:\j\workspace\windows-TGrou---0d2c9ac4\src\System.Data.SqlClient\pkg\System.Data.SqlClient.pkgproj]
error : System.Data.SqlClient should support API version 4.4.0.0 on .NETCoreApp,Version=v2.0/win7-x86 but 
System.Data.SqlClient\netstandard-Windows_NT\System.Data.SqlClient.dll was found to support 4.6.0.0. [D:\j\workspace\windows-TGrou---0d2c9ac4\src\System.Data.SqlClient\pkg\System.Data.SqlClient.pkgproj]
... x 50
error : System.Data.SqlClient should support API version 4.4.0.0 on UAP,Version=v10.0.16300/win10-arm-aot but 
System.Data.SqlClient\uap-Windows_NT\System.Data.SqlClient.dll was found to support 4.6.0.0. [D:\j\workspace\windows-TGrou---0d2c9ac4\src\System.Data.SqlClient\pkg\System.Data.SqlClient.pkgproj]
```

**Comment by Eilon** [comment]
FYI marking this as blocking-partner because this is required for 3.0 Preview 1 for ASP.NET Core.

**Comment by joperezr** [comment]
> FYI marking this as blocking-partner because this is required for 3.0 Preview 1 for ASP.NET Core.

Not really. What is really blocking ASP.NET now is a different PR due to signing issues #33387

**Comment by danmosemsft** [comment]
Bunch of package validation failures.

**Pull request merged**